### DISCUSSION TITLE
Can we have a Micro-frontend with one part Next.js and another Create-React-App

### DISCUSSION THREAD

**Discussion by Suedo** [vercel/next.js#14020] [Help] [answered]
As a learning practice.. I thought of making a crude medium clone. My app is of two parts basically:

1. The public facing pages are made with Next.js part for static generation and SEO, 
2. while the stuff behind an auth-wall, like dashboard, the "editor" for creating a post etc, is made with Create-React-App and rendered client side.

I am now wondering how to tie them up together. I'm pretty new to Next.js and unsure how the routing will work. Any ideas ?

**Comment by jamesmosier** [comment, marked as answer]
You would have to serve them as completely separate applications, they can't be combined. Each `page` in Next is its own entry point, so there would be no way to render create-react-app within that page.

Also FWIW Next contains all the stuff that create-react-app does plus a lot more 😄 you could setup some authentication rules for individual pages within Next and leave some public.

**Reply by Suedo** [reply]
I was reading that there is some issue with MobX + Next. hence the thought.. if MobX and Next work nicely, i have no need for create-react-app :)

Also, something you said:

> there would be no way to render create-react-app within that page

I am not looking to render a CRA component inside a Next Page. I was thinking of having the entire app that sits behind and auth-flow be C-R-A. Separate routes and all. Does that make any difference ?

**Reply by jamesmosier** [reply]
There is an example of using [mobx](https://github.com/vercel/next.js/tree/canary/examples/with-mobx) and Next.

And sure, you could just run 2 separate applications and then just link between them. One way would be to have 2 servers running each one of your applications and then you'd have to manage the routing somewhere with something like nginx to direct users traffic for each one to the proper server.

**Reply by Suedo** [reply]
Thanks for the overview :) Much clearer now :)

**Reply by yogeshwar607** [reply]
@Suedo you implemented this ?

**Reply by vikyij** [reply]
Hi @Suedo 
I'm trying to implement something similar, I have an existing react app and I want to create a Nextjs app for my landing pages because of SEO. Did you later implement this?
